# rosparam 사용 시 클래스 내에서 파라미터를 받는 변수 타입과 일치하는 값을 적어주어야 함
# ex) int : 3, float : 3.0
# 파라미터 순서랑 클래스 내에서 파라미터 변수 선언할 때 순서랑 맞춰줄 것?

# 라이다 높이
# kitti : 1.
# nuscenes : 1.84m
# songdo : 1.9m

Public:
  map: "/map/songdo.json"
  lidar_frame: "hesai_lidar"
  ins_frame: "gps"
  target_frame: "ego_car"
  world_frame: "world"

Cloud_Segmentation:
  lidar_settings:
    V_SCAN: 64 # Pandar 64 or OT 128
    H_SCAN: 1800
    resolution_x: 0.2 # Calculated as 360.0 / H_SCAN
    resolution_y: 0.3125 # Calculated as 40.0 / V_SCAN
    ang_bottom: 25

  crop:
    max:
      x: 80.0
      y: 80.0
      z: 0.1
    min:
      x: -80.0
      y: -80.0
      z: -1.7 # kitti , nuscenes 1.6, songdo 1.7
    crop_ring:
      enabled: false
      ring: 2
    crop_intensity:
      enabled: false
      intensity: 10.0
    crop_hd_map:
      radius: 2.0
  
  ground_removal:
    fp_distance: 7.0

  downsampling:
    leaf_size:
      x: 0.2
      y: 0.2
      z: 0.2

  clustering:
    filter:
      min_size:
        x: 0.2
        y: 0.3
        z: 0.3
      max_size:
        x: 13.0
        y: 13.0
        z: 4.0
    adaptive:
      min_size: 10
      max_size: 2000
      min_tolerance: 0.5
      max_tolerance: 1.0
      max_region_distance: 100
      number_region: 10
      min_leaf_size: 0.2
      max_leaf_size: 0.4
      thresh_iou: 0.1
    L_shape_fitting:
      projection_range: 0.3

Tracking:
  integration:
    mode: 0 # 0: integration, 1: cluster, 2: deep
    thresh_iou: 0.1
  crop_hd_map:
    radius: 2.6
  track:
    invisibleCnt: 5
    deque:
      number_velocity: 4
      number_orientation: 10
      thresh_velocity: 13.0 # m/s
      thresh_orientation: 13.0 # degree